RHIROS-531 ROS BE Sequence and DB schema diagram

srdeotarse opened the pull request:
## ROS BE Sequence and DB schema diagram :boom:

[RHIROS-531](https://issues.redhat.com/browse/RHIROS-531) UML diagram for ROS Backend repo

Added small desc about ROS and ROS BE processing flow in README.

Also added ROS DB Schema and ROS BE Sequence Diagram in docs folder.

Feel free to remove this section from PR description once done.

## Why do we need this change? :thought_balloon:

Please include the __context of this change__ here.

## Documentation update? :memo:

- [x] Yes
- [ ] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

r14chandra commented on docs/ROS BE Sequence Diagram.png:
Could you also include `puml` file so that it would be easy for anyone to update the diagram later in the future if needed?

r14chandra replied:
I see you have included the links. I wanted to see if it is possible to add something as a file to make future changes easier.

srdeotarse replied:
We can edit the UML markup for both the diagrams in [PlantUML Web editor](https://www.plantuml.com/plantuml/uml/SyfFKj2rKt3CoKnELR1Io4ZDoSa70000) by copying the UML diagram link from README and pasting it in PlantUML Web Server. Decode URL and make changes as required.

srdeotarse replied:
@kgaikwad  I understand your point. Maybe, I can save the UML source code in docs folder for both the UML diagrams ( DB Schema and Sequence diagram).

r14chandra commented on docs/ROS DB Schema Diagram.png:
Same for the ERD diagram.

r14chandra commented on docs/ROS BE Sequence Diagram.png:
I see some missing fields for the insights engine events message like `platform_metadata`. Could you verify them?
If possible, could you please format the JSON to make it more readable?

srdeotarse replied:
As input field contains same fields as host payload (that includes platform_metadata), I have mentioned ``<same as host payload>`` to focus on ``results`` field

r14chandra replied:
But IIRC, `host` field doesn't contain `platform_metadata`.

srdeotarse replied:
@r14chandra  I meant to say, the ``platform.inventory.events`` message is same as ``platform.engine.results``  message ``"input":`` field value. Therefore I have written ``"input": <same as host payload>`` to focus on ``"results":`` value

r14chandra reviewed (changes requested):
Thanks @srdeotarse for the PR!
I have posted a few suggestions.

r14chandra reviewed (changes requested):
@srdeotarse, we have the current schema enhancement for the `Suggested Instance Type` feature. The PR is open for QE to test but it is approved. IMO we should include that as well.
Let me know your thoughts.

kgaikwad commented on README.md:
```diff
@@ -2,7 +2,17 @@
+
+## How it works
+
+![UML](https://www.plantuml.com/plantuml/png/fLDDR-f65DtxLrnD5zs0-rwxGn5w8Q5Lf2GW72Ab0Lc3vmAZs3FMV92WrF-zTx0TaJgmw6E1FEUU-ttk3-kuSRx88oTTZZ2R933YsGig0LEZCxHMgWtycUlNg2Icp6J9bOFjIaQeo6SFPSxTMfiYbcf7a3Rx6CE7_UJ0fB9oitLseL1jf4BMyQ_nsA3rENclz1kmeteO_78R8enoZn7GxtFuDZdKoHUghClNXdNPTIgeKttcXiufTIIuuojkiQxXUiGYwha_FDdj0BRQEagpppKNqKbUV_Z5ABfGawaxbm6qt7wcpKSFqGawfJlJxBrsP0ZHGA_XPCPtgg_qZWtWuhZ9OM-XcDkNI0 ...
```
In this diagram, at engine result processor component, you mentioned `Create / Update perf_profile for updated\nsystem in PerformanceProfile table`.
Though method name says create_or_update, every time it creates fresh performance profile record.
could you modify the statement something like - `Create Performance Profile record for a system in  table`?

kgaikwad reviewed (changes requested):
Added two small inline suggestions.

srdeotarse commented:
> @srdeotarse, we have the current schema enhancement for the `Suggested Instance Type` feature. The PR is open for QE to test but it is approved. IMO we should include that as well. Let me know your thoughts.

@r14chandra Yes, I reviewed the PR. I will add the new columns related to `Suggested Instance type` feature to DB schema diagram.

kgaikwad reviewed (approved):
Looks good to me. Thank you @srdeotarse!

@r14chandra, do you have any additional comments?

r14chandra commented:
@srdeotarse One unaddressed comment above. Let me know if we want to address or close it.

Why tests are failing here 🤔

srdeotarse commented:
@r14chandra @kgaikwad I think tests are also failing for new dependabot PRs as well, something related to RedisCache error

Also, I have addressed your comment. Hope that helps.

r14chandra reviewed (approved):
The PR looks good @srdeotarse !
